kmp-maps/core/src/iosMain/kotlin/com/swmansion/kmpmaps/core/MKGeoJsonRenderedLayer.kt

```diff
@@ -0,0 +1,381 @@
+package com.swmansion.kmpmaps.core
+
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.Foundation.NSArray
+import platform.Foundation.NSDictionary
+import platform.Foundation.NSJSONSerialization
+import platform.Foundation.NSNumber
+import platform.Foundation.NSString
+import platform.Foundation.NSUTF8StringEncoding
+import platform.Foundation.dataUsingEncoding
+import platform.Foundation.enumerateKeysAndObjectsUsingBlock
+import platform.MapKit.*
+import platform.UIKit.UIColor
+
+/**
+ * Style for a GeoJSON LineString/Polyline rendered on Apple Maps (MapKit).
+ *
+ * @param color Stroke color (UIKit UIColor).
+ * @param width Stroke width in screen points.
+ */
+public data class AppleGeoJsonLineStyle(val color: UIColor, val width: Double)
+
+/**
+ * Style for a GeoJSON Polygon rendered on Apple Maps (MapKit).
+ *
+ * @param strokeColor Outline color (UIKit UIColor).
+ * @param strokeWidth Outline width in screen points.
+ * @param fillColor Optional fill color (UIKit UIColor). Use null to disable fill.
+ */
+public data class AppleGeoJsonPolygonStyle(
+    val strokeColor: UIColor,
+    val strokeWidth: Double,
+    val fillColor: UIColor?,
+)
+
+/**
+ * Style for a GeoJSON Point rendered on Apple Maps (MapKit).
+ *
+ * @param visible Controls marker visibility.
+ */
+public data class AppleGeoJsonPointStyle(val visible: Boolean = true)
```

Comment:
For these three, wouldn’t it be preferable to use `AppleMaps` as the prefix, in line with the existing Apple Maps–specific classes (e.g., `AppleMapsPointOfInterestCategory`)?